arm/Microsoft.Network/privateDnsZones/virtualNetworkLinks/deploy.bicep

```diff
@@ -0,0 +1,51 @@
+@description('Required. Private DNS zone name.')
+param privateDnsZoneName string
+
+@description('Optional. The name of the virtual network link.')
+param name string = last(split(virtualNetworkId, '/'))
+
+@description('Optional. The location of the PrivateDNSZone. Should be global.')
+param location string = 'global'
+
+@description('Optional. Tags of the resource.')
+param tags object = {}
+
+@description('Optional. Is auto-registration of virtual machine records in the virtual network in the Private DNS zone enabled?')
+param registrationEnabled bool = false
+
+@description('Required. Link to another virtual network Id.')
+param virtualNetworkId string
+
+@description('Optional. Customer Usage Attribution id (GUID). This GUID must be previously registered')
+param cuaId string = ''
+
+module pid_cuaId '.bicep/nested_cuaId.bicep' = if (!empty(cuaId)) {
+  name: 'pid-${cuaId}'
+  params: {}
+}
+
+resource privateDnsZone 'Microsoft.Network/privateDnsZones@2018-09-01' existing = {
+  name: privateDnsZoneName
+}
+
+resource virtualNetworkLink 'Microsoft.Network/privateDnsZones/virtualNetworkLinks@2018-09-01' = {
+  name: name
+  location: location
+  tags: tags
+  properties: {
+    registrationEnabled: registrationEnabled
+    virtualNetwork: {
+      id: virtualNetworkId
+    }
+  }
+  parent: privateDnsZone
+}
```

Comment:
Check this with Alex. Not sure we have decided on this being the structure we want to go with yet. Before that decision is made, I would suggest that we keep it to the current way of doing this. @MrMCake What's your take on this? Should we not stay with the current agreement, while having the 2 module that POC the other approaches leaving the rest in line?

Current way of implementing this: <https://github.com/Azure/ResourceModules/blob/main/arm/Microsoft.Compute/virtualMachines/extensions/deploy.bicep#L47>

Issue describing the change, where no decision has been made, before we do the restructuring of modules:
https://github.com/Azure/ResourceModules/issues/480

Reply:
Hey @MariusStorhaug, should be fine here. We used the same in for example AutomationAccount. On this level it does not require any special treatment. Hence I'd keep it here

Reply:
Ok!